### PR TITLE
ENH: implement partial_fit for SciKeras models

### PR DESCRIPTION
**Pull request by stsievert**
**What does this PR implement?**
It implements and tests `partial_fit` for `scikeras.BaseWrapper` models. It also exposes a `warm_start` parameter in `BaseWrapper.fit`.

**Reference issues/PRs**
This will be useful for https://github.com/dask/dask-ml/issues/696.

### CODE REVIEW

**Comment by adriangb** [comment]
This is super cool! Definitely something I was thinking about. I will take a look later tonight.

**Comment by stsievert** [comment]
> Do you think it would be necessary to support serialization + partial fits (i..e partial fit -> serialize -> partial fit, etc.)?

I'd like to see that because it'd be really useful in Dask-ML's model selection. I'll implement a test for that.

**Comment by adriangb** [comment]
@stsievert mind giving me write access to your fork so that I can try and work out the test failures? Thanks!

File: scikeras/wrappers.py [hunk 1]
```diff
@@ -453,7 +458,7 @@ def _pre_process_X(X):
         extra_args = dict()
         return X, extra_args
 
-    def fit(self, X, y, sample_weight=None, **kwargs):
+    def fit(self, X, y, sample_weight=None, warm_start=True, **kwargs):
```
Comment (adriangb) [inline scikeras/wrappers.py:461]: This should be `warm_start=False`. This is what was causing the `OutputShape` tests to fail.

File: tests/test_wrappers.py [hunk 2]
```diff
@@ -1199,3 +1200,58 @@ def test_pprint(self):
...
+
+        X, y = data.data[:100], data.target[:100]
+        clf.partial_fit(X, y)
+
+        clf2 = pickle.loads(pickle.dumps(clf))
+        clf2.partial_fit(X, y)
+        assert len(clf.history_["loss"]) == 1
+        assert len(clf2.history_["loss"]) == 2
+        assert np.allclose(clf.history_["loss"][0], clf2.history_["loss"][0])
+        assert clf2.history_["loss"][1] <= clf2.history_["loss"][0]
```
Comment (adriangb) [inline tests/test_wrappers.py:1244]: It does not seem like the loss reliably decreases. Do you feel strongly that this is a good test, or can we do without/come up with an alternative @stsievert?

File: scikeras/wrappers.py (hunk 1 above)
Comment (stsievert) [reply scikeras/wrappers.py]: D'oh. Thanks for the catch.

File: tests/test_wrappers.py (hunk 2 above)
Comment (stsievert) [reply tests/test_wrappers.py]: I've implemented a better test. I'm interested in making sure the optimizer and weights are hooked together. Now, I test that the weights have changed.

File: tests/test_wrappers.py [hunk 3]
```diff
@@ -1199,3 +1199,137 @@ def test_pprint(self):
...
+        assert 1000 <= sum(n_weights) <= 2000
+        assert 200 <= np.mean(n_weights) <= 300
+        assert max(n_weights) >= 1000
+        rel_errors = [
+            np.linalg.norm(w1 - w2) / np.linalg.norm((w1 + w2) / 2)
+            for w1, w2 in zip(weights1, weights2)
+        ]
+        assert len(rel_errors) == 4
+        assert any(x > 0.5 for x in rel_errors)
+        assert all(0.01 < x for x in rel_errors)
```
Comment (adriangb) [inline tests/test_wrappers.py:1317]: This looks great! Could you add some comments to clarify what the constants are and such?

File: tests/test_wrappers.py (hunk 2 above)
Comment (adriangb) [reply tests/test_wrappers.py]: I saw, left a comment on there. Thank you!

File: tests/test_wrappers.py (hunk 3 above)
Comment (stsievert) [reply tests/test_wrappers.py]: Added.

**Comment by adriangb** [comment]
Everything looks good. Ready to merge @stsievert ?

**Comment by stsievert** [comment]
Looks good to me. Thanks @adriangb!

**Comment by stsievert** [comment]
Give me a sec – looks like I committed an extra file.

**Comment by stsievert** [comment]
I'd appreciate if you could "squash and merge". That'll keep the git history clean – I accidentally committed `test-output.xml` (not to mention my commit messages).

**Comment by adriangb** [comment]
Got it. Thanks again for this PR! Please let me know if there's anything I can do to facilitate contributions.

**Comment by stsievert** [comment]
> Please let me know if there's anything I can do to facilitate contributions.

To be honest, I'm not sure how you could have better facilitated this contribution:

* Google wasn't involved, so I didn't have to sign anything.
* This is a small third party package, so development can happen pretty smoothly (no lengthy reviews, not many breaking tests, etc).
* You even debugged for me!

> ping me if you need a release from this package for dask/dask-ml#696.

Let's wait for https://github.com/dask/dask-ml/pull/699, which depends on some other issues.

**Comment by stsievert** [comment]
Could you make a new release and push to PyPI? I'm testing on CI now, which runs `pip install scikeras`.

**Comment by adriangb** [comment]
Do you need that asap, or within the next couple of days? Could you install
from source until then? Just so I can try to collect 1-2 more improvements
before releasing.

**Comment by stsievert** [comment]
I don't need it immediately – I'm installing from the master branch now in Dask-ML's CI pipeline. It would be nice to have in a couple days; I think https://github.com/dask/dask-ml/pull/699 will be ready to merge then.

**Comment by adriangb** [comment]
Got it. I'll try to match up the timelines.
